xbmc/guilib/handlers/sources/GUISourcesAnnouncementHandler.cpp

```diff
@@ -0,0 +1,42 @@
+/*
+ *  Copyright (C) 2024 Team Kodi
+ *  This file is part of Kodi - https://kodi.tv
+ *
+ *  SPDX-License-Identifier: GPL-2.0-or-later
+ *  See LICENSES/README.md for more information.
+ */
+
+#include "GUISourcesAnnouncementHandler.h"
+
+#include "GUIUserMessages.h"
+#include "ServiceBroker.h"
+#include "guilib/GUIComponent.h"
+#include "guilib/GUIWindowManager.h"
+#include "interfaces/AnnouncementManager.h"
+
+CGUISourcesAnnouncementHandler::CGUISourcesAnnouncementHandler()
+{
+  CServiceBroker::GetAnnouncementManager()->AddAnnouncer(this);
+}
+
+CGUISourcesAnnouncementHandler::~CGUISourcesAnnouncementHandler()
+{
+  CServiceBroker::GetAnnouncementManager()->RemoveAnnouncer(this);
+}
+
+void CGUISourcesAnnouncementHandler::Announce(ANNOUNCEMENT::AnnouncementFlag flag,
+                                              const std::string& sender,
+                                              const std::string& message,
+                                              const CVariant& data)
+{
+  // We are only interested in sources changes
+  if ((flag & ANNOUNCEMENT::Sources) == 0)
+    return;
+
+  if (message == "OnAdded" || message == "OnRemoved" || message == "OnUpdated")
+  {
+    CGUIMessage message(GUI_MSG_NOTIFY_ALL, 0, 0, GUI_MSG_UPDATE_PATH);
+    message.SetStringParam(data["path"].asString());
```

Comment:
Would it make sense to introduce a new `CGUIMessage` ctor to save the call to `SetStringParam` ?

Reply:
I'm seeing many of those two-liners in our code base.

<img width="382" alt="image" src="https://github.com/user-attachments/assets/c7b3ea92-1195-4233-a249-73989fe4d553">

Reply:
Alright.